### DISCUSSION TITLE
Option to force synchronization: ignore ETag etc.

### DISCUSSION THREAD

**Discussion by ltguillaume** [bitfireAT/icsx5#392] [Ideas]
It seems that ICSx5 always checks for `ETag`/`If-None-Match` and `Last-Modified`/`If-Modified-Since`, so there isn't a way to force a synchronization.

There are several situations in which the local calendar storage can become corrupted, or where the user has to manually clear the Calendar Storage data via the Android settings in order to resolve other issues. After this, however, ICSx5 will _never_ sync these calendars again, until the remote source has been changed.

I propose a new option called "Force sync" in the overflow menu that avoids checking `Etag` etc. headers and always synchronizes.

**Comment by ArnyminerZ** [comment]
Hi, this is already a feature, but is only used when editing a subscription. For example, changing its name, so if you modify the name of the subscription, and save it, the next synchronization will ignore all saved eTags and modification dates, and always fetch the data from the server.

However, I agree that there should be a more intuitive, explained or at least user-friendly way of doing this.

Do you think that it will be intuitive enough that the pull-to-refresh action is the "normal" sync, and if the overflow menu option is chosen, this "forced" sync will be run? Maybe modify the text to something like "Sync now (forced)" or something like that? @sunkup also what do you think?

**Reply by ltguillaume** [reply]
Only if the feature is renamed to "Force Sync" or "Sync now (forced)", yes.

**Reply by ArnyminerZ** [reply]
I think I prefer "Force Sync", a bit shorter

**Comment by ArnyminerZ** [comment]
Followup: #397
